pkg/kubelet/rkt/container_id.go

```diff
@@ -41,6 +42,14 @@ func buildContainerID(c *containerID) kubecontainer.ContainerID {
 	}
 }
 
+func apiPodToNetPluginContainerID(pod *api.Pod) kubecontainer.ContainerID {
+	return buildContainerID(&containerID{string(pod.UID), pod.Name})
+}
+
+func runtimePodToNetPluginContainerID(pod *kubecontainer.Pod) kubecontainer.ContainerID {
+	return buildContainerID(&containerID{string(pod.ID), pod.Name})
```

Comment:
I'm not seeing this, not sure why:

```
I0505 10:17:07.380677   26358 rkt.go:1063] Calling network plugin kubenet to setup pod for hi-openshift_default(6e3c09c0-12d4-11e6-a99d-54ee754bd54c)
W0505 10:17:07.380697   26358 container_id.go:51] $$$$$$$ apiPod container ID &rkt.containerID{uuid:"6e3c09c0-12d4-11e6-a99d-54ee754bd54c", appName:"hi-openshift"}
W0505 10:17:07.380717   26358 container_id.go:52] $$$$$$$ apiPod kube container ID container.ContainerID{Type:"rkt", ID:"6e3c09c0-12d4-11e6-a99d-54ee754bd54c:hi-openshift"}
W0505 10:17:07.380735   26358 kubenet_linux.go:222] ^^^^^ kubenet: container id container.ContainerID{Type:"rkt", ID:"6e3c09c0-12d4-11e6-a99d-54ee754bd54c:hi-openshift"}
W0505 10:17:07.380747   26358 rkt.go:1547] ##### container id container.ContainerID{Type:"rkt", ID:"6e3c09c0-12d4-11e6-a99d-54ee754bd54c:hi-openshift"}
W0505 10:17:07.380760   26358 rkt.go:1548] ##### id &rkt.containerID{uuid:"6e3c09c0-12d4-11e6-a99d-54ee754bd54c", appName:"hi-openshift"}
W0505 10:17:07.380772   26358 rkt.go:1553] ##### string UID "6e3c09c0-12d4-11e6-a99d-54ee754bd54c"
W0505 10:17:07.380782   26358 kubenet_linux.go:227] ^^^^^ kubenet: netns path "/var/run/netns/k8s_6e3c09c0-12d4-11e6-a99d-54ee754bd54c"
I0505 10:17:07.380792   26358 kubenet_linux.go:307] Kubenet: using netns path /var/run/netns/k8s_6e3c09c0-12d4-11e6-a99d-54ee754bd54c
I0505 10:17:07.380797   26358 kubenet_linux.go:308] Kubenet: using podns path default
I0505 10:17:07.380805   26358 kubenet_linux.go:320] Calling cni plugins to add container to network with cni runtime: &{ContainerID:6e3c09c0-12d4-11e6-a99d-54ee754bd54c:hi-openshift NetNS:/var/run/netns/k8s_6e3c09c0-12d4-11e6-a99d-54ee754bd54c IfName:eth0 Args:[]}

```

But I agree that it's very, very confusing with all the IDs flying around.  I'll try to clean it up.